### ISSUE TITLE
feat: implement no-emphasis-as-heading rule (MD036)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#128]
## Overview

Implement the `no-emphasis-as-heading` rule (equivalent to markdownlint MD036 / remark-lint `no-emphasis-as-heading`), tracked in #76.

## Problem

Writers sometimes use bold or italic text on a line by itself as a visual heading substitute:

```markdown
**Introduction**

Some text here.
```

This is semantically incorrect — screen readers and document outlines won't recognise it as a heading.

## Proposed rule key

`no-emphasis-as-heading`

## Detection logic

Flag a line when **all** of the following are true:

1. The trimmed line consists entirely of a single bold (`**…**`, `__…__`) or italic (`*…*`, `_…_`) span.
2. The inner text does **not** end with sentence-ending punctuation (see list below).
3. The line is not inside a fenced code block or inline code span.

### Punctuation list

Mirrors the default punctuation list of markdownlint MD036, including full-width equivalents for CJK writing systems:

| Type | Characters |
|---|---|
| ASCII | `. , ; : ! ?` |
| Full-width / CJK | `。 、 ； ： ！ ？` |

### Examples

| Input | Result |
|---|---|
| `**Section Title**` | violation |
| `__Section Title__` | violation |
| `*Section Title*` | violation |
| `_Section Title_` | violation |
| `**はじめに**` | violation |
| `*概要*` | violation |
| `**This is a sentence.**` | OK (ends with `.`) |
| `**Note:**` | OK (ends with `:`) |
| `**これは文章です。**` | OK (ends with `。`) |
| `**注意！**` | OK (ends with `！`) |
| `This is **important** text.` | OK (not the entire line) |
| `` `**bold**` `` | OK (inside inline code) |

## Fix

Replace bare emphasis with an ATX heading:

```markdown
## Introduction

Some text here.
```

## References

- markdownlint MD036: https://github.com/DavidAnson/markdownlint/blob/main/doc/md036.md
- remark-lint `no-emphasis-as-heading`
- Parent issue: #76
